### DISCUSSION TITLE
Why is Sha2 prioritized over Sha3 as base hash algorithm?

### DISCUSSION THREAD

**Discussion by nitinklohar** [DMTF/libspdm#2615] [Q&A] [answered]
In the libspdm_get_response_algorithms function, priority table for hash algorithm (https://github.com/DMTF/libspdm/blob/9bc69639f112be9d5bd8a4aa93cd0fe07d70361a/library/spdm_responder_lib/libspdm_rsp_algorithms.c#L76) prioritizes Sha2 over Sha3. Is there any specific reason for this? Shouldn't Sha3 be prioritized over Sha2 as it is more secure?

**Comment by steven-bellock** [comment, marked as answer]
>Is there any specific reason for this?

No. See #121. In particular, if a Responder supports more than one algorithm then it should be up to the Integrator to determine priority. However, anecdotally, most Responder devices only support one algorithm family (SHA2 XOR SHA3), and even then only one algorithm within that family, and so that decision typically doesn't have to be made.

>as it is more secure

You mean it's more secure as it does not have to worry about length extension attacks? While SHA2 is susceptible to length extension attacks, algorithms that incorporate SHA2, such as HMAC, are able to mitigate that issue.

**Reply by nitinklohar** [reply]
Thanks. This helps.

**Comment by jyao1** [comment]
> as it is more secure?

Would you please clarify which document says: SHA3 is more secure than SHA2?
